s3/pkg/datastore/yig/storage/gc.go

```diff
@@ -0,0 +1,281 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package storage
+
+import (
+	"context"
+	"errors"
+	"runtime"
+	"sync"
+	"time"
+
+	. "github.com/opensds/multi-cloud/s3/error"
+	"github.com/opensds/multi-cloud/s3/pkg/datastore/yig/meta/types"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	GC_OBJECT_LIMIT_NUM = 10000
+)
+
+type GcMgr struct {
+	// context to cancel the operations.
+	ctx        context.Context
+	cancelFunc context.CancelFunc
+	yig        *YigStorage
+	loopTime   int64
+	wg         sync.WaitGroup
+}
+
+func (gm *GcMgr) Start() {
+	// query the current available cpus
+	threadNum := runtime.GOMAXPROCS(0)
```

Comment:
How many routines we will start? Will that use too much cpu resource?

Reply:
@sfzeng thanks for your questions here, if multi-cloud process is running in the docker container, this call will return all the available cpus despite there is a cpu restriction for this container, and https://github.com/uber-go/automaxprocs is used to fix this problem. Because that 3rd lib will introduce so many files in this pr, could i open an issue to track this problem and fix this problem in a separate pr? thanks a lot.

Reply:
OK, please open an issue, and fix in later PR, thanks !

Reply:
@sfzeng thanks a lot, i have created issue https://github.com/opensds/multi-cloud/issues/741 to track this problem you found. i will submit a separate pr for this issue soon.